### DISCUSSION TITLE
API for traversing all fields for data-type conversion on submit?

### DISCUSSION THREAD

**Discussion by msageryd** [react-hook-form/react-hook-form#7642] [Q&A]
I'm investigating a move of my data-type conversion from onChange to onSubmit.

@bluebill1049 suggested that I'd leave all values as strings until the form is submitted, and do the conversion at submit. Do I need to traverse the data manually and match the fields with my field definition? 

My field definitions does not represent the actual object structure, but instead uses dot-notation for easy field registration in RHF. Unfortunately, this flat structure does not match the submitted data structure.

It would be great if `handleSubmit` allowed for pre-processing the data in a flat manner, i.e. dot-notation.

I could traverse the data and build the flat field-names as I go, using dot-notation. But this process is obviously already handled within RHF, so this feels like inventing the wheel again.

Example, simplified field definition:
```
fieldDefinition: {
  fieldA: {
    dataType: 'integer'
  },
  "deep.data": {
    dataType: 'number'
  }
}
```

The data I receive from `handleSubmit`:
```
{
  fieldA: '11',
  deep: {
    data: '1.12'
  }
}
```

**Comment by bluebill1049** [comment]
we don't have functionality, you may have to build this at your app level.
